chain/chain/src/test_utils.rs

```diff
@@ -166,25 +197,67 @@ impl KeyValueRuntime {
         // We cannot do any reasonable validations of it in test_utils.
         state.insert(StateRoot::default(), kv_state);
         state_size.insert(StateRoot::default(), data_len);
+        #[cfg(feature = "protocol_feature_chunk_only_producers")]
+        assert_eq!(block_producers.len(), chunk_only_producers.len());
+        let mut validators = HashMap::new();
+        let block_producers: Vec<Vec<_>> = block_producers
+            .iter()
+            .map(|account_ids| {
+                account_ids
+                    .iter()
+                    .map(|account_id| {
+                        let stake = ValidatorStake::new(
+                            account_id.clone(),
+                            SecretKey::from_seed(KeyType::ED25519, account_id.as_ref())
+                                .public_key(),
+                            1_000_000,
+                        );
+                        validators.insert(account_id.clone(), stake.clone());
+                        stake
+                    })
+                    .collect()
+            })
+            .collect();
+        #[cfg(feature = "protocol_feature_chunk_only_producers")]
+        let chunk_only_producers: Vec<Vec<Vec<_>>> = chunk_only_producers
+            .into_iter()
+            .map(|v| {
+                v.into_iter()
+                    .map(|account_ids| {
+                        account_ids
+                            .into_iter()
+                            .map(|account_id| {
+                                let stake = ValidatorStake::new(
+                                    account_id.clone(),
+                                    SecretKey::from_seed(KeyType::ED25519, account_id.as_ref())
+                                        .public_key(),
+                                    1_000_000,
+                                );
+                                validators.insert(account_id, stake.clone());
+                                stake
+                            })
+                            .collect()
+                    })
+                    .collect()
+            })
+            .collect();
         KeyValueRuntime {
             store,
             tries,
-            validators: validators
-                .iter()
-                .map(|account_ids| {
-                    account_ids
-                        .iter()
-                        .map(|account_id| {
-                            ValidatorStake::new(
-                                account_id.clone(),
-                                SecretKey::from_seed(KeyType::ED25519, account_id.as_ref())
-                                    .public_key(),
-                                1_000_000,
-                            )
-                        })
-                        .collect()
+            #[cfg(feature = "protocol_feature_chunk_only_producers")]
+            validators_by_valset: zip(block_producers, chunk_only_producers)
```

Comment:
```suggestion
            validators_by_valset: std::iter::zip(block_producers, chunk_only_producers)
```

* no need to import itertools here
* usually, I'd `use std::iter`, but with fully qualified syntax we can avoid cfg.